java/src/main/native/include/jni_utils.hpp

```diff
@@ -790,19 +760,43 @@ inline void jni_cuda_check(JNIEnv *const env, cudaError_t cuda_status) {
+    }                                                                                              \
+    std::string n_msg = e.what() == nullptr ? "" : e.what();                                       \
+    jstring j_msg = env->NewStringUTF(n_msg.c_str());                                              \
+    jint e_code = static_cast<jint>(e.error_code());                                               \
+    jclass ex_class = env->FindClass(class_name);                                                  \
+    jmethodID ctor_id = env->GetMethodID(ex_class, "<init>", "(Ljava/lang/String;I)V");            \
+    if (ctor_id == NULL) {                                                                         \
+      return ret_val;                                                                              \
+    }                                                                                              \
+    jobject cuda_error = env->NewObject(ex_class, ctor_id, j_msg, e_code);                         \
+    env->Throw((jthrowable)cuda_error);                                                            \
+    if (is_halt) {                                                                                 \
+      cudf::jni::throw_jni_exception("CUDA ERROR");                                                \
+    }                                                                                              \
+    return ret_val;                                                                                \
+  } while (0)
+
+#define JNI_CUDA_TRY(ret_val, call)                                                                \
+  do {                                                                                             \
     cudaError_t internal_cuda_status = (call);                                                     \
     if (cudaSuccess != internal_cuda_status) {                                                     \
-      /* Clear the last error so it does not propagate.*/                                          \
-      cudaGetLastError();                                                                          \
-      jthrowable jt = cudf::jni::cuda_exception(env, internal_cuda_status);                        \
-      if (jt != NULL) {                                                                            \
-        env->Throw(jt);                                                                            \
-      }                                                                                            \
+      cudf::detail::throw_cuda_error(internal_cuda_status, __FILE__, __LINE__);                    \
       return ret_val;                                                                              \
     }                                                                                              \
-  }
+  } while (0)
+
+#define JNI_CUDA_CHECK(cuda_status)                                                                \
```

Comment:
This is semantically equivalent to the previous macro, and we can just use `CUDF_CUDA_TRY` instead.

Reply:
Removed.